hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hudi/ProvidesHoodieConfig.scala

```diff
@@ -284,8 +286,23 @@ trait ProvidesHoodieConfig extends Logging {
       }
     }
 
+    val (recordMergeMode, recordMergeStrategy) = if (payloadClassName.equals(classOf[ValidateDuplicateKeyPayload].getCanonicalName)) {
+      (RecordMergeMode.CUSTOM.name(), HoodieRecordMerger.PAYLOAD_BASED_MERGER_STRATEGY_UUID)
+    } else {
+      (RecordMergeMode.EVENT_TIME_ORDERING.name(), HoodieRecordMerger.DEFAULT_MERGER_STRATEGY_UUID)
+    }
+
+    if (tableConfig.getPayloadClass.equals(classOf[DefaultHoodieRecordPayload].getCanonicalName) &&
+        tableConfig.getRecordMergeMode.equals(RecordMergeMode.EVENT_TIME_ORDERING)) {
+      tableConfig.clearValue(HoodieTableConfig.PAYLOAD_CLASS_NAME)
```

Comment:
Is this modified table config only used here? Does it have any side effect?

Reply:
This is the place where we needed to have the custom handling. combineOptions takes the tableconfig over the overriding opts. But now we are always setting the table configs, so the overriding opts will never be used here. So we discussed and decided that the best solution was just to clear out the tableconfig here if we think that the user just used the default

Reply:
I was asking whether the modified table config is also used elsewhere causing behavior change.  So looks like it only determines the write config to be returned, which is good.